src/queue/queue-data-structure.js

```diff
@@ -5,25 +5,48 @@ class Queue {
   }
 
   canEnqueue() {
-    // ... your code goes here
+    if (this.queueControl.length === this.MAX_SIZE) {
+      return false;
+    }
+    if (this.queueControl.length < this.MAX_SIZE) {
+      return true;
+    }
   }
 
   isEmpty() {
-    // ... your code goes here
+    if (this.queueControl.length === 0) {
+      return true;
+    }
+    return false;
   }
 
   enqueue(item) {
-    // ... your code goes here
+    if (this.canEnqueue() === true) {
+      this.queueControl.push(item);
+      return this.queueControl;
+    }
+    if (this.canEnqueue() === false) {
+      throw new Error("QUEUE_OVERFLOW");
+    }
   }
 
   dequeue() {
-    // ... your code goes here
+    if (this.isEmpty() === false) {
+      return this.queueControl.shift();
+    }
+    if (this.isEmpty() === true) {
+      throw new Error("QUEUE_UNDERFLOW");
+    } // ... your code goes here
   }
 
   display() {
```

Comment:
Si te das cuenta, estás devolviendo lo mismo que hay en `this.queueControl`. Con retornarlo ya es suficiente:
``` Javascript
return this.queueControl
```